tests/Aspire.Dashboard.Tests/Integration/Playwright/PlaywrightTestsBase.cs

```diff
@@ -11,20 +12,39 @@ public class PlaywrightTestsBase : IClassFixture<DashboardServerFixture>, IClass
     public DashboardServerFixture DashboardServerFixture { get; }
     public PlaywrightFixture PlaywrightFixture { get; }
 
+    private readonly ITestOutputHelper _output;
     private IBrowserContext? _context;
+    private readonly Guid _id;
 
-    public PlaywrightTestsBase(DashboardServerFixture dashboardServerFixture, PlaywrightFixture playwrightFixture)
+    public PlaywrightTestsBase(DashboardServerFixture dashboardServerFixture, PlaywrightFixture playwrightFixture, ITestOutputHelper output)
     {
         DashboardServerFixture = dashboardServerFixture;
         PlaywrightFixture = playwrightFixture;
+        _output = output;
+        _id = Guid.NewGuid();
     }
 
-    public async Task RunTestAsync(Func<IPage, Task> test)
+    public async Task RunTestAsync(Func<IPage, Task<bool>> setup, Func<IPage, Task> test)
     {
         var page = await CreateNewPageAsync();
         try
         {
-            await test(page);
+            var shouldRunTest = true;
+
+            try
+            {
+                await setup(page);
+            }
+            catch (PlaywrightException e)
+            {
+                _output.WriteLine($"Test setup failed: {e}");
+                await TakeScreenshotAsync("dashboard-fail.png", page);
```

Comment:
We should let the test fail, and the screenshot can be used for debugging the failure. I don't think we need to split this into two steps.
If the failure is because we need to wait longer then a longer timeout should be fine. If it's not that then maybe something is failing on the page.